file-formats/aifp/type/zif_aff_aifp_v1.intf.abap

```diff
@@ -0,0 +1,293 @@
+    END OF ty_main.
+
+  CONSTANTS:
+    BEGIN OF co_check_scenario,
+      "! <p class="shorttext">Simple Field Check</p>
+      "! Simple field check
+      simple     TYPE ty_check_scenario VALUE 'SIMPLE',
+      "! <p class="shorttext">Advanced Field Check</p>
+      "! Advanced field check
+      advanced   TYPE ty_check_scenario VALUE 'ADVANCED',
+      "! <p class="shorttext">Simple Field Check and Database Check</p>
+      "! Simple field check  and database check
+      simpledb   TYPE ty_check_scenario VALUE 'SIMPLEDB',
+      "! <p class="shorttext">Advanced Field Check and Database Check</p>
+      "! Advanced field check and database check
+      advanceddb TYPE ty_check_scenario VALUE 'ADVANCEDDB',
+      "! <p class="shorttext">Database Check</p>
+      "! Database check
+      database   TYPE ty_check_scenario VALUE 'DATABASE',
+      "! <p class="shorttext">Manual Implementation</p>
+      "! Manual implementation
+      implement  TYPE ty_check_scenario VALUE 'IMPLEMENT',
+    END OF co_check_scenario.
+  CONSTANTS:
+    BEGIN OF co_simple_field_check,
+      "! <p class="shorttext">Empty</p>
+      "! Empty
+      empty              TYPE ty_simple_field_check VALUE 'I',
+      "! <p class="shorttext">Not Empty</p>
+      "! Not empty
+      not_empty          TYPE ty_simple_field_check VALUE 'N',
+      "! <p class="shorttext">Empty or '0'</p>
+      "! Empty or '0'
+      empty_or_zero      TYPE ty_simple_field_check VALUE '0',
+      "! <p class="shorttext">Numeric (Integer)</p>
+      "! Numeric (Integer)
+      numeric_integer    TYPE ty_simple_field_check VALUE '1',
+      "! <p class="shorttext">Numeric or Empty</p>
+      "! Numeric or empty
+      numeric_empty      TYPE ty_simple_field_check VALUE '2',
+      "! <p class="shorttext">Numeric (Integer/Negative)</p>
+      "! Numeric (Integer/Negative)
+      numeric_negativ    TYPE ty_simple_field_check VALUE '3',
+      "! <p class="shorttext">Numeric (Comma/Negative)</p>
+      "! Numeric (Comma/Negative)
+      numeric_comma      TYPE ty_simple_field_check VALUE '4',
+      "! <p class="shorttext">Numeric (Dot/Negative)</p>
+      "! Numeric (Dot/Negative)
+      numeric_dot        TYPE ty_simple_field_check VALUE '5',
+      "! <p class="shorttext">Only Characters A-Z</p>
+      "! Only characters A-Z
+      only_chars_capital TYPE ty_simple_field_check VALUE 'A',
+      "! <p class="shorttext">Only Characters a-z</p>
+      "! Only characters a-z
+      only_chars_lower   TYPE ty_simple_field_check VALUE 'a',
+      "! <p class="shorttext">Only Characters A-Z, a-z</p>
+      "! Only characters A-Z, a-z
+      only_chars         TYPE ty_simple_field_check VALUE 'Z',
+      "! <p class="shorttext">Alphanumeric with special chars and German Umlauts</p>
+      "! Alphanumeric with special chars and German Umlauts
+      x                  TYPE ty_simple_field_check VALUE 'X',
+    END OF co_simple_field_check.
+  CONSTANTS:
+    BEGIN OF co_operator,
```

Comment:
in interface zif_aff_types_v1 we have constant co_option.
Do you want to use the constants from zif_aff_types_v1=>co_option?
But zif_aff_types_v1=>co_option does not have your last 3 options(contains_any_character,not_contains_any_character and matches). Also for "equal" and "non_equal" the values are different than yours.

Reply:
Actually, we're uncertain about supporting the last three operators in ADT at this time, and are leaning towards not doing so. Would it introduce a incompatible change if we altered the type and added these additional operators in the future?
The values for equal and non_equal was wrong and matches with your type.